app/controllers/devices_controller.rb

```diff
@@ -40,7 +40,7 @@ def device_params
       user: current_user,
       token: params[:token],
       platform: params[:platform],
-      app_bundle: params[:app_bundle]
+      app_integration: AppIntegration.find_by(app_bundle: params[:app_bundle])
```

Comment:
what happens if this returns nil? can it be nil?

Reply:
It can't be nil because it's a `belongs_to` association that [isn't optional](https://stackoverflow.com/a/35775686/3462026).

The effect of this will be that requests to _create new devices_ will fail unless they include a valid `app_bundle`, which makes sense because we don't want Devices created not-associated to App Integrations (they would be effectively useless).